### ISSUE TITLE
sqlfluff-templater-dbt ImportError still happening even when installing 

### ISSUE THREAD

**Issue by ezeray** [sqlfluff/sqlfluff#4374]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Upgraded to newest dbt core version `dbt-core==1.4.1`, the most recent version. I tried to run `sqlfluff` to lint a `.sql` file and got the following error:

```
ImportError: cannot import name 'CompilationException' from 'dbt.exceptions' 
```

I know that this was raised in other issues (#4315 #4345) and that it was addressed in pull request #4317 , but the version being installed is incorrect.

### Expected Behaviour

The linter to run succesfully as it did before upgrading the packages.

### Observed Behaviour

Did some immediate checking on the versions of the packages, and followed the trace of the error. Found that this was already tackled

In `templater.py` module in `sqlfluff-templater-dbt` [(source)](https://github.com/sqlfluff/sqlfluff/blob/main/plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py), the import is done as follows (line 18 to 27):

```python
try:
    from dbt.exceptions import (
        CompilationException as DbtCompilationException,
        FailedToConnectException as DbtFailedToConnectException,
    )
except ImportError:
    from dbt.exceptions import (
        CompilationError as DbtCompilationException,
        FailedToConnectError as DbtFailedToConnectException,
    )
```

However, when I check the `templater.py` module in my environment's `.../lib/python3.10/site-packages/sqlfluff-templater-dbt`, the code looks as follows:

```python
from dbt.exceptions import (
    CompilationException as DbtCompilationException,
    FailedToConnectException as DbtFailedToConnectException,
)
```

Considering that this was fixed in PR #4317 , I believe that it might be related to the package versioning, but honestly not sure of how that would happen.

A quick way to get around this is to just copy the results of the most updated `templater.py` and copy it into the site-packages, but that's not scalable and quite annoying. Hoping there's an easy fix and that this helps anybody that encounters the same issue.

### How to reproduce

Steps to reproduce. Please note that I tried this using pyton version `python==3.11` and got the same result. Also cleaned my cache and did a `--no-cache` install of all packages.

```bash
python3.10 -m venv dbt-env

source dbt-env/bin/activate

pip install --upgrade pip

# install dbt-core and check that it runs correctly
pip install dbt-core
dbt --version

# Core:
#  - installed: 1.4.1
#  - latest:    1.4.1 - Up to date!

# install sqlfluff and check that it runs correctly
pip install sqlfluff
sqlfluff --version

# sqlfluff, version 1.4.5

# install sqlfluff-templater-dbt
pip install sqlfluff-templater-dbt

sqlfluff --version

# Traceback (most recent call last):
# ...
# ImportError: cannot import name 'CompilationException' from 'dbt.exceptions' (.../dbt-env/lib/python3.10/site-packages/dbt/exceptions.py)
```

### Dialect

`dialect=snowflake`

### Version

```
sqlfluff=1.4.5
dbt-core=1.4.1
python=3.10.9
pip==23.0
```

### Configuration

.sqlfluff

```
[sqlfluff]
templater = dbt

[sqlfluff:templater:jinja]
apply_dbt_builtins = True
```

.sqlfluffignore

```
target/
macros/
dbt_packages/
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by kokorin** [comment]
Use this:
```
sqlfluff==2.0.0a4
sqlfluff-templater-dbt==2.0.0a4
```
It worked for me

**Comment by ezeray** [comment]
Thanks @kokorin, using this package version solved the issue.

Will close the ticket since it's solved by using versions
```
sqlfluff==2.0.0a4
sqlfluff-templater-dbt==2.0.0a4
```

**Comment by asheone** [comment]
Sorry, but it doesn't seem like it's working in either version `2.0.0a4` or `2.0.0a5`. Are you sure it's fixed in the source? @kokorin @ezeray ?
